public/app/themes/clarity/inc/admin/users/protect-key-pages.php

```diff
@@ -0,0 +1,119 @@
+ * It filters user capabilities to prevent non-admin users from editing or deleting these key pages.
+ */
+class ProtectKeyPages
+{
+    // Key templates for pages that only admins can edit.
+    const KEY_TEMPLATES = [
+        'page_blog.php',
+        'page_blogroll.php',
+        'page_deleteuser.php',
+        'page_events.php',
+        'page_guidance_and_support_index.php',
+        'page_home.php',
+        'page_news.php',
+    ];
+
+    // Key slugs for pages that only admins can edit.
+    const KEY_SLUGS = [
+        'agency-switcher',
+    ];
+
+    function __construct()
+    {
+        add_filter('user_has_cap', [$this, 'filterUserCapabilities'], 10, 3);
+    }
+
+    /**
+     * Checks is a post or page is limited to editing by admin only.
+     * e.g. the page that has been slected as the front page, or a page with the template of page_blog.php
+     * 
+     * @param int $object_id The ID of the post or page to check.
+     * @return bool True if the post is a special post that only admins can edit, false otherwise.
+     */
+    public function isKeyPage(int $object_id): bool
+    {
+        $post = get_post($object_id);
+
+        if (!$post || 'page' !== $post->post_type) {
+            return false; // Post does not exist, or is not a page
+        }
+
+        $post_template = get_page_template_slug($post->ID);
+
+        // Check if the page uses a special template.
+        if (in_array($post_template, self::KEY_TEMPLATES, true)) {
+            return true;
+        }
+
+        // Check if the page has a special slug.
+        if (in_array($post->post_name, self::KEY_SLUGS, true)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Dynamically filter user capabilities, to block page edits or deletions.
+     *
+     * Filter the capabilities base on:
+     * - if the user is an administrator
+     * - the requested capability
+     * - the object ID (typically a post or page ID)
+     * 
+     * @param bool[]   $allcaps Array of key/value pairs where keys represent a capability name
+     *                          and boolean values represent whether the user has that capability.
+     * @param string[] $caps    Required primitive capabilities for the requested capability.
+     * @param array    $args {
+     *     Arguments that accompany the requested capability check.
+     *
+     *     @type string    $0 Requested capability.
+     *     @type int       $1 Concerned user ID.
+     *     @type mixed  ...$2 Optional second and further parameters, typically object ID.
+     * }
+     * @param WP_User  $user    The user object.
+     * @return bool[]           Filtered array of capabilities.
+     */
+    public function filterUserCapabilities($allcaps, $cap, $args)
+    {
+        // Check if the user is an administrator
+        if (isset($allcaps['administrator']) && $allcaps['administrator']) {
```

Comment:
This can be tested locally by commenting out this block.

Or you can, switch users to a non-admin user.